pkg/scheduler/algorithm/priorities/ready.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package priorities
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/klog"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+	"math"
+)
+
+// ReadyPod contains information to calculate ready pod score.
+type ReadyPod struct {
+}
+
+// NewReadyPodPriority creates a ReadyPod.
+func NewReadyPodPriority() PriorityFunction {
+	readyPod := &ReadyPod{}
+	return readyPod.CalculateReadyPodPriority
+}
+func (r *ReadyPod) isPodReady(pod *v1.Pod) bool {
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type == "Ready" && cond.Status == "True" {
+			return true
+		}
+	}
+	return false
+}
+
+// CalculateReadyPodPriority favors nodes with lesser count of non ready pods.
+// Calcuated on a scale of 0-10 where 10 is highest priority.
+func (r *ReadyPod) CalculateReadyPodPriority(pod *v1.Pod, nodeNameToInfo map[string]*schedulernodeinfo.NodeInfo, nodes []*v1.Node) (framework.NodeScoreList, error) {
+	totalPodsNotReady := 0
+	nodeToNonReadyPods := make(map[string]int)
+	// Get count of non ready pods for each node.
+	for _, node := range nodes {
+		nonReadyPods := 0
+		nodeinfo := nodeNameToInfo[node.Name]
+		pods := nodeinfo.Pods()
+		for _, p := range pods {
+			isPodReady := r.isPodReady(p)
+			klog.V(10).Infof("node: %v, pod: %v, ready: %v", node.Name, p.ObjectMeta.Name, isPodReady)
+			if !r.isPodReady(p) {
```

Comment:
you already have a variable `isPodReady`, so you could use that instead of looping through all the pod conditions again. Or probably just collapse this into one check